ui/main/src/app/modules/externaldevicesconfiguration/editModal/signal-mappings-modal.component.ts

```diff
@@ -0,0 +1,121 @@
+/* Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * See AUTHORS.txt
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of the OperatorFabric project.
+ */
+
+import {Component, Input, OnInit} from '@angular/core';
+import {AbstractControl, AsyncValidatorFn, FormControl, FormGroup, ValidationErrors, Validators} from '@angular/forms';
+import {NgbActiveModal} from '@ng-bootstrap/ng-bootstrap';
+import {ExternalDevicesService} from 'app/business/services/notifications/external-devices.service';
+import {SignalMapping} from '@ofModel/external-devices.model';
+import {Observable, of} from 'rxjs';
+
+@Component({
+    selector: 'of-signal-mappings-modal',
+    templateUrl: './signal-mappings-modal.component.html',
+    styleUrls: ['./externaldevicesconfiguration-modal.component.scss']
+})
+export class SignalMappingsModalComponent implements OnInit {
+    signalMappingsForm: FormGroup;
+
+    @Input() row: any;
+
+    selectedDevice;
+    signalMappings: SignalMapping[];
+    isLoadingSignalMappings: boolean;
+
+    constructor(private activeModal: NgbActiveModal) {}
+
+    ngOnInit() {
+        const uniqueDeviceIdValidator = [];
+        if (!this.row) {
+            uniqueDeviceIdValidator.push(this.uniqueDeviceIdValidatorFn());
+        }
+        this.signalMappingsForm = new FormGroup({
+            id: new FormControl<string | null>('', [Validators.required], uniqueDeviceIdValidator),
+            alarm: new FormControl<string | null>('', [Validators.required]),
+            action: new FormControl<string | null>('', [Validators.required]),
+            information: new FormControl<string | null>('', [Validators.required]),
+            compliant: new FormControl<string | null>('', [Validators.required])
+        });
+
+        this.isLoadingSignalMappings = true;
+        ExternalDevicesService.queryAllSignalMappings().subscribe((signalMappings) => {
+            this.signalMappings = signalMappings;
+            this.isLoadingSignalMappings = false;
+        });
+
+        if (this.row) {
+            this.signalMappingsForm.patchValue(this.row, {onlySelf: true});
+            this.selectedDevice = this.row.deviceId;
+        }
+    }
+
+    uniqueDeviceIdValidatorFn(): AsyncValidatorFn {
+        return (control: AbstractControl): Observable<ValidationErrors> => {
+            const err: ValidationErrors = {uniqueDeviceIdViolation: true};
+            return this.isUniquedeviceId(this.signalMappingsForm.controls['id'].value) ? of(null) : of(err);
+        };
+    }
+
+    isUniquedeviceId(mappingId: string): boolean {
+        if (mappingId && this.signalMappings.findIndex((dev) => dev.id === mappingId) >= 0) return false;
+        else return true;
```

Comment:
Return directly the opposite of the findIndex() call, no need to have an if... else ...